public/app/features/manage-dashboards/components/PublicDashboardListTable.tsx

```diff
@@ -0,0 +1,112 @@
+import { css } from '@emotion/css';
+import React, { FC, useState } from 'react';
+import useAsync from 'react-use/lib/useAsync';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { getBackendSrv } from '@grafana/runtime';
+import { Link, ButtonGroup, LinkButton, Icon, Tag, useStyles2 } from '@grafana/ui';
+import { getConfig } from 'app/core/config';
+
+export interface ListPublicDashboardResponse {
+  uid: string;
+  accessToken: string;
+  dashboardUid: string;
+  title: string;
+  isEnabled: boolean;
+}
+
+export const listPublicDashboardsUrl = () => {
+  return `/api/dashboards/public`;
+};
+export const getPublicDashboards = async () => {
+  const resp: ListPublicDashboardResponse[] = await getBackendSrv().get(listPublicDashboardsUrl());
+  return resp.sort((a, b) => Number(b.isEnabled) - Number(a.isEnabled));
+};
+
+export const viewPublicDashboardUrl = (accessToken: string): string => {
+  return `${getConfig().appUrl}public-dashboards/${accessToken}`;
+};
+
+function getStyles(theme: GrafanaTheme2) {
+  return {
+    link: css`
+      color: ${theme.colors.primary.text};
+      text-decoration: underline;
+      margin-right: ${theme.spacing()};
+    `,
+  };
+}
+
+export const ListPublicDashboardTable: FC = () => {
+  const styles = useStyles2(getStyles);
+  const [publicDashboards, setPublicDashboards] = useState<ListPublicDashboardResponse[]>([]);
+
+  useAsync(async () => {
+    const publicDashboards = await getPublicDashboards();
+    setPublicDashboards(publicDashboards);
+  }, [setPublicDashboards]);
+
+  function renderEnabledTag(pd: ListPublicDashboardResponse) {
+    if (pd.isEnabled) {
+      return <Tag name="enabled" colorIndex={20} />;
+    } else {
+      return <Tag name="disabled" colorIndex={15} />;
+    }
+  }
+
+  function renderViewLink(pd: ListPublicDashboardResponse) {
+    let title = pd.isEnabled ? 'View public dashboard' : 'Public dashboard is disabled';
+    return (
+      <LinkButton
+        href={viewPublicDashboardUrl(pd.accessToken)}
+        fill="text"
+        title={title}
+        target="_blank"
+        disabled={!pd.isEnabled}
+      >
+        <Icon name="external-link-alt" />
+      </LinkButton>
+    );
+  }
+
+  function renderConfigLink(pd: ListPublicDashboardResponse) {
+    let url = `/d/${pd.dashboardUid}?shareView=share`;
+    return (
+      <LinkButton fill="text" href={url} title="Configure public dashboard">
+        <Icon name="cog" />
+      </LinkButton>
+    );
+  }
+
+  return (
+    <div className="page-action-bar">
+      <table className="filter-table">
```

Comment:
this is a global class which gives you just width 100%. I would implement a class in this file in the styles object and don't depend in this one

Reply:
Agreed- Let's add this to refactor ticket